docs/src/main/tut/Job.md

```diff
@@ -1 +1,65 @@
-# Job
+# Job\[A\]
+
+On a `TypedDataset` all operations are lazy. An operation will either return a new 
+transformed `TypedDataset` or it will return a `Job[A]`, where A is the result of running a
+non-lazy computation in Spark. `Job` serves several functions: 
+
+- Makes all operations on a `TypedDataset` lazy, which gives less surprises compared to having
+few operations being lazy and other being strict
+- Allows the programmer to be more explicit when an expensive blocking operation needs to happen
+- Allows for Spark jobs to be lazily sequenced using monadic composition via for-comprehension
+- Provides an obvious place where you can annotate/name your Spark jobs to make it easier
+to track different parts of your application in the Spark UI
+
+The toy example showcases the use of for-comprehension to explicitly sequences Spark Jobs. 
+First we calculate the size of the `TypedDataset` and then we collect to the driver
+exactly 20% of its elements: 
+
+```tut:invisible
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.sql.SparkSession
+import frameless.functions.aggregate._
+import frameless.TypedDataset
+
+val conf = new SparkConf().setMaster("local[*]").setAppName("frameless repl").set("spark.ui.enabled", "false")
+val spark = SparkSession.builder().config(conf).appName("REPL").getOrCreate()
+implicit val sqlContext = spark.sqlContext
+spark.sparkContext.setLogLevel("WARN")
+
+import spark.implicits._
+```
+
+```tut:book
+val ds = TypedDataset.create(1 to 20)
+
+val countAndTakeJob = for(c <- ds.count(); sample <- ds.take((c/5).toInt)) yield sample
```

Comment:
break lines:

    for {
      c      <- ds.count() // rename c to count?
      sample <- ds.take((c / 5).toInt)
    } yield sample